java/RSPEC-4635.java

```diff
@@ -0,0 +1,8 @@
+class Substring {
```

Comment:
RSPEC-2176: Class has same name as parent
<!-- lookout footnote separator -->
_Comment made by the analyzer sonarcheck_analyzer._